Can't build with -fno-common (default with llvm 11 and GCC 10)

Hi!
I'm unable to get libobjc2 to build with -fno-common, which is the default with llvm 11 and GCC 10.
When building on FreeBSD 13-CURRENT with llvm 11, I'm getting the following error on linking:
```
: && /usr/bin/cc -fPIC -O2 -pipe  -fno-strict-aliasing -Xclang -fexceptions -Xclang -fobjc-exceptions -O3 -O2 -pipe  -fno-strict-aliasing  -pthread -shared -Wl,-soname,libobjc.so.4.6 -o libobjc.so.4.6 CMakeFiles/objc.dir/alias_table.c.o CMakeFiles/objc.dir/block_to_imp.c.o CMakeFiles/objc.dir/caps.c.o CMakeFiles/objc.dir/category_loader.c.o CMakeFiles/objc.dir/class_table.c.o CMakeFiles/objc.dir/dtable.c.o CMakeFiles/objc.dir/encoding2.c.o CMakeFiles/objc.dir/hooks.c.o CMakeFiles/objc.dir/ivar.c.o CMakeFiles/objc.dir/loader.c.o CMakeFiles/objc.dir/mutation.m.o CMakeFiles/objc.dir/protocol.c.o CMakeFiles/objc.dir/runtime.c.o CMakeFiles/objc.dir/sarray2.c.o CMakeFiles/objc.dir/selector_table.c.o CMakeFiles/objc.dir/sendmsg2.c.o CMakeFiles/objc.dir/eh_personality.c.o CMakeFiles/objc.dir/legacy.c.o CMakeFiles/objc.dir/abi_version.c.o CMakeFiles/objc.dir/statics_loader.c.o CMakeFiles/objc.dir/block_trampolines.S.o CMakeFiles/objc.dir/objc_msgSend.S.o CMakeFiles/objc.dir/eh_trampoline.s.o CMakeFiles/objc.dir/NSBlocks.m.o CMakeFiles/objc.dir/Protocol2.m.o CMakeFiles/objc.dir/associate.m.o CMakeFiles/objc.dir/blocks_runtime.m.o CMakeFiles/objc.dir/properties.m.o CMakeFiles/objc.dir/gc_none.c.o CMakeFiles/objc.dir/arc.mm.o CMakeFiles/objc.dir/objcxx_eh.cc.o  -lcxxrt && :
ld: error: duplicate symbol: __objc_msg_forward2
>>> defined at hooks.c
>>>            CMakeFiles/objc.dir/hooks.c.o:(__objc_msg_forward2)
>>> defined at sendmsg2.c
>>>            CMakeFiles/objc.dir/sendmsg2.c.o:(.data+0x10)

ld: error: duplicate symbol: __objc_msg_forward3
>>> defined at hooks.c
>>>            CMakeFiles/objc.dir/hooks.c.o:(__objc_msg_forward3)
>>> defined at sendmsg2.c
>>>            CMakeFiles/objc.dir/sendmsg2.c.o:(.data+0x8)

ld: error: duplicate symbol: _objc_selector_type_mismatch
>>> defined at hooks.c
>>>            CMakeFiles/objc.dir/hooks.c.o:(_objc_selector_type_mismatch)
>>> defined at sendmsg2.c
>>>            CMakeFiles/objc.dir/sendmsg2.c.o:(.bss+0x0)

ld: error: duplicate symbol: _objc_selector_type_mismatch2
>>> defined at hooks.c
>>>            CMakeFiles/objc.dir/hooks.c.o:(_objc_selector_type_mismatch2)
>>> defined at sendmsg2.c
>>>            CMakeFiles/objc.dir/sendmsg2.c.o:(.data+0x18)

ld: error: duplicate symbol: objc_proxy_lookup
>>> defined at hooks.c
>>>            CMakeFiles/objc.dir/hooks.c.o:(objc_proxy_lookup)
>>> defined at sendmsg2.c
>>>            CMakeFiles/objc.dir/sendmsg2.c.o:(.data+0x0)
cc: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```
I've tried to dig through the source code, but I'm unable to figure out exactly how to fix it.  All of these are in `objc/hooks.h`, where they are prefixed with `OBJC_HOOK`, which is defined to `OBJC_PUBLIC extern`. However, when `hooks.h` is included in `hooks.c`, `OBJC_HOOK` is defined to nothing, meaning that the symbols are no longer extern. These symbols are also defined in sendmsg2.c as `IMP (*__objc_msg_forward2)(id, SEL) = forward2;` (as an example), and I believe this is causing the duplicate symbols.
It might be that it is enough to declare them `extern` in sendmsg2.c as well, but I can't figure out what `IMP` is in the definition above, and I'm unsure what the implication of having `extern IMP` or `IMP extern` would be.

--- a/objc/hooks.h
+++ b/objc/hooks.h
@@ -35,17 +35,17 @@ OBJC_HOOK void (*_objc_load_callback)(Class cls, struct objc_category *category)
  * The hook used for fast proxy lookups.  This takes an object and a selector
  * and returns the instance that the message should be forwarded to.
  */
-OBJC_HOOK id (*objc_proxy_lookup)(id receiver, SEL op);
+OBJC_PUBLIC extern id (*objc_proxy_lookup)(id receiver, SEL op);
 /**
  * New runtime forwarding hook.  This is no longer used, but is retained to
  * prevent errors at link time.
  */
-OBJC_HOOK struct objc_slot *(*__objc_msg_forward3)(id, SEL) OBJC_DEPRECATED;
+OBJC_PUBLIC extern struct objc_slot *(*__objc_msg_forward3)(id, SEL) OBJC_DEPRECATED;
 /**
  * Forwarding hook.  Takes an object and a selector and returns a method that
  * handles the forwarding.
  */
-OBJC_HOOK IMP (*__objc_msg_forward2)(id, SEL);
+OBJC_PUBLIC extern IMP (*__objc_msg_forward2)(id, SEL);
 /**
  * Hook defined for handling unhandled exceptions.  If the unwind library
  * reaches the end of the stack without finding a handler then this hook is
@@ -67,13 +67,13 @@ OBJC_HOOK Class (*_objc_class_for_boxing_foreign_exception)(int64_t exceptionCla
  * receiver.  This should return the slot to use instead, although it may throw
  * an exception or perform some other action.
  */
-OBJC_HOOK IMP (*_objc_selector_type_mismatch2)(Class cls, 
+OBJC_PUBLIC extern IMP (*_objc_selector_type_mismatch2)(Class cls, 
        SEL selector, struct objc_slot2 *result);
 /**
  * Legacy hook for when selector types do not match.  This is only called
  * `_objc_selector_type_mismatch2` is not installed.
  */
-OBJC_HOOK struct objc_slot *(*_objc_selector_type_mismatch)(Class cls,
+OBJC_PUBLIC extern struct objc_slot *(*_objc_selector_type_mismatch)(Class cls,
        SEL selector, struct objc_slot *result) OBJC_DEPRECATED;
 
 /**

